sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImplTest.java

```diff
@@ -0,0 +1,136 @@
+
+    distance = CassandraServiceImpl.distance(cluster, 100L, 10L);
+    assertEquals(new BigInteger("170141183460469231731687303715884105639"), distance);
+  }
+
+  @Test
+  public void testRingFractionWithRandomPartitioner() throws Exception {
+    Cluster cluster = createClusterMock(RANDOM_PARTITIONER);
+    List<CassandraServiceImpl.TokenRange> tokenRanges = new ArrayList<>();
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(0, 100, 0, 99));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(1, 100, 100, 199));
+    BigDecimal ringFraction = CassandraServiceImpl.getRingFraction(cluster, tokenRanges);
+
+    System.out.println(ringFraction);
+  }
+
+  @Test
+  public void testDistanceWithMurmur3Partitioner() throws Exception {
+    Cluster cluster = createClusterMock(MURMUR3_PARTITIONER);
+    BigInteger distance = CassandraServiceImpl.distance(cluster, 10L, 100L);
+
+    assertEquals(new BigInteger("90"), distance);
+
+    distance = CassandraServiceImpl.distance(cluster, 100L, 10L);
+    assertEquals(new BigInteger("18446744073709551525"), distance);
+  }
+
+  private long getEstimatedSizeBytes(String partitioner) {
+    Cluster cluster = createClusterMock(partitioner);
+    List<CassandraServiceImpl.TokenRange> tokenRanges = new ArrayList<>();
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(0, 10000000, 0, 99999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(1, 10000000, 100000000, 199999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(2, 10000000, 200000000, 299999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(3, 10000000, 300000000, 399999999));
+    tokenRanges.add(new CassandraServiceImpl.TokenRange(4, 10000000, 400000000, 499999999));
+    return CassandraServiceImpl.getEstimatedSizeBytes(cluster, tokenRanges);
+  }
+
+  @Test
+  public void testEstimatedSizeBytesWithMurmur3Partitioner() throws Exception {
+    assertEquals(100000000, getEstimatedSizeBytes(MURMUR3_PARTITIONER));
+  }
+
+  @Test
+  public void testEstimatedSizeBytesWithRandomPartitioner() throws Exception {
+    assertEquals(100000000, getEstimatedSizeBytes(RANDOM_PARTITIONER));
+  }
+
+  private List<CassandraIO.CassandraSource> testSplitIntoBundles(String partitionerName)
+      throws Exception {
+    Cluster cluster = createClusterMock(partitionerName);
+    CassandraServiceImpl.Partitioner partitioner = new CassandraServiceImpl.Partitioner(cluster);
+    CassandraServiceImpl service = new CassandraServiceImpl();
+    CassandraIO.Read spec = CassandraIO.read().withKeyspace("beam").withTable("test");
+    return service.split(spec, 1000, 10000, partitioner);
+  }
+
+  @Test
+  public void testSplitIntoBundlesWithMurmur3Partitioner() throws Exception {
+    List<CassandraIO.CassandraSource> sources = testSplitIntoBundles(MURMUR3_PARTITIONER);
+    assertEquals(10, sources.size());
+    for (CassandraIO.CassandraSource source : sources) {
+      LOG.info(source.splitQuery);
```

Comment:
We need to test that the queries are correct. Otherwise it's very easy to have a buggy implementation that will still pass the test.